lib/acts_as_indexed/class_methods.rb

```diff
@@ -134,25 +137,59 @@ def search_index(query, find_options={}, options={})
 
       return part_query if options[:ids_only]
 
-      with_scope :find => find_options do
-        # Doing the find like this eliminates the possibility of errors occuring
-        # on either missing records (out-of-sync) or an empty results array.
-        records = find(:all, :conditions => [ "#{table_name}.#{primary_key} IN (?)", part_query])
-
-        if find_options.include?(:order)
-         records # Just return the records without ranking them.
+      # New 2023 way to find the records.
+      conditions = ["#{table_name}.#{primary_key} IN (?)", part_query]
+      query_options = find_options.slice(:limit, :offset, :order)
+      records = where(conditions)
 
-         else
-           # Results come back in random order from SQL, so order again.
-           ranked_records = ActiveSupport::OrderedHash.new
-           records.each do |r|
-             ranked_records[r] = @query_cache[query][r.id]
-           end
+      if find_options[:conditions]
+        records = records.where(find_options[:conditions])
+      end
+      if find_options[:joins]
+        records = records.joins(find_options[:joins])
+      end
+      if find_options[:includes]
+        records = records.includes(find_options[:includes])
+      end
+      if find_options[:limit]
+        records = records.limit(find_options[:limit])
+      end
+      if find_options[:offset]
+        records = records.offset(find_options[:offset])
+      end
+      if find_options[:order]
+        records = records.order(find_options[:order])
+        return records # Just return the records without ranking them.
+      end
 
-           sort(ranked_records.to_a).map{ |r| r.first }
-         end
+      # Results come back in random order from SQL, so order again.
+      ranked_records = ActiveSupport::OrderedHash.new
```

Comment:
This can be a hash literal, as AS:OrderedHash was only relevant in Ruby 1.8 and earlier which did not maintain Hash insert order.

Reply:
Sure I'll change this to a regular hash